<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Wei Xu, Meiling Hao, Yi Zhu" />

<meta name="date" content="2019-06-27" />

<title>xlink</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' || rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">xlink</h1>
<h4 class="author">Wei Xu, Meiling Hao, Yi Zhu</h4>
<h4 class="date">2019-06-27</h4>



<div id="introduction" class="section level1">
<h1>1. Introduction</h1>
<p><strong>xlink</strong> from <a href="https://github.com/qiuanzhu/xlink">github</a> is a package for the unified partial likelihood approach for X-chromosome association on time-to-event/ continuous/ binary outcomes. The expression of X-chromosome undergoes three possible biological processes: X-chromosome inactivation (XCI), escape of the X-chromosome inactivation (XCI-E), and skewed X-chromosome inactivation (XCI-S). Although these expressions are included in various predesigned genetic variation chip platforms, the X-chromosome has generally been excluded from the majority of genome-wide association studies analyses; this is most likely due to the lack of a standardized method in handling X-chromosomal genotype data. To analyze the X-linked genetic association for time-to-event outcomes with the actual process unknown, we propose a unified approach of maximizing the partial likelihood over all of the potential biological processes. The proposed method can be used to infer the true biological process and derive unbiased estimates of the genetic association parameters.</p>
<div id="reference" class="section level2">
<h2>Reference:</h2>
<blockquote>
<p>“Xu, Wei, and Meiling Hao.”A unified partial likelihood approach for X‐chromosome association on time‐to‐event outcomes.&quot; Genetic epidemiology 42.1 (2018): 80-94.&quot; (<a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/gepi.22097">via</a>)</p>
</blockquote>
</div>
</div>
<div id="installation" class="section level1">
<h1>2. Installation</h1>
<p>You can install <strong>xlink</strong> from [github]((<a href="https://github.com/qiuanzhu/xlink" class="uri">https://github.com/qiuanzhu/xlink</a>):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;devtools&quot;</span>)
<span class="kw">install_github</span>(<span class="st">&quot;qiuanzhu/xlink&quot;</span>)</code></pre></div>
</div>
<div id="examples" class="section level1">
<h1>3. Examples</h1>
<p>In the following examples, we choose the <strong>model</strong> is “survival” model, which could also applied to “linear” model for continuous response and “binary” for fitting logistic regression model.</p>
<div id="select-significant-snps-from-xci-or-xci-e-model-type" class="section level2">
<h2>3.1 Select significant SNPs from XCI or XCI-E model type:</h2>
<p>In the sample data with 10 SNPs and 4 clinic covariates,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;xlink&quot;</span>)
<span class="kw">head</span>(Rdata)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="right">ID</th>
<th align="right">OS</th>
<th align="right">OS_time</th>
<th align="right">gender</th>
<th align="right">Age</th>
<th align="right">Smoking</th>
<th align="right">Treatment</th>
<th align="right">snp_1</th>
<th align="right">snp_2</th>
<th align="right">snp_3</th>
<th align="right">snp_4</th>
<th align="right">snp_5</th>
<th align="right">snp_6</th>
<th align="right">snp_7</th>
<th align="right">snp_8</th>
<th align="right">snp_9</th>
<th align="right">snp_10</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0.0335</td>
<td align="right">1</td>
<td align="right">44.3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0.0424</td>
<td align="right">1</td>
<td align="right">76.9</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0.6435</td>
<td align="right">1</td>
<td align="right">53.7</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">4</td>
<td align="right">1</td>
<td align="right">0.3548</td>
<td align="right">0</td>
<td align="right">63.1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">5</td>
<td align="right">1</td>
<td align="right">0.0306</td>
<td align="right">0</td>
<td align="right">29.2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">6</td>
<td align="right">1</td>
<td align="right">0.3050</td>
<td align="right">1</td>
<td align="right">77.5</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>If the Model type is chosen to be <strong>XCI</strong> and threshold for <strong>MAF_v</strong> is set to be 0.05, the output for snp_1 with coefficient, P value and loglikelihood information</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Covars&lt;-<span class="kw">c</span>(<span class="st">&quot;Age&quot;</span>,<span class="st">&quot;Smoking&quot;</span>,<span class="st">&quot;Treatment&quot;</span>)
SNPs&lt;-<span class="kw">c</span>(<span class="st">&quot;snp_1&quot;</span>,<span class="st">&quot;snp_2&quot;</span>)
output&lt;-<span class="kw">xlink_fit</span>(<span class="dt">os=</span><span class="st">&quot;OS&quot;</span>,<span class="dt">ostime=</span><span class="st">&quot;OS_time&quot;</span>,<span class="dt">snps=</span>SNPs,<span class="dt">gender=</span><span class="st">&quot;gender&quot;</span>,<span class="dt">covars=</span>Covars, <span class="dt">option =</span><span class="kw">list</span>(<span class="dt">type=</span><span class="st">&quot;XCI&quot;</span>,<span class="dt">MAF_v=</span><span class="fl">0.05</span>),<span class="dt">model=</span><span class="st">&quot;survival&quot;</span>,<span class="dt">data =</span> Rdata)</code></pre></div>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">Hazard Ratio</th>
<th align="left">Confidence Interval (95%)</th>
<th align="right">P Value</th>
<th align="left">MAF</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>snp_1</td>
<td align="right">1.6392</td>
<td align="left">[1.3895,1.9339]</td>
<td align="right">0.0000000</td>
<td align="left">0.20625</td>
</tr>
<tr class="even">
<td>gender</td>
<td align="right">0.9529</td>
<td align="left">[0.7557,1.2015]</td>
<td align="right">0.6834041</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td>Age</td>
<td align="right">1.0228</td>
<td align="left">[1.0155,1.0301]</td>
<td align="right">0.0000000</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td>Smoking</td>
<td align="right">1.3001</td>
<td align="left">[1.0172,1.6617]</td>
<td align="right">0.0360414</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td>Treatment</td>
<td align="right">1.2356</td>
<td align="left">[0.9781,1.561]</td>
<td align="right">0.0760446</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<table>
<thead>
<tr class="header">
<th align="right">Baseline</th>
<th align="right">Full model</th>
<th align="right">Loglik ratio</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">-1493.808</td>
<td align="right">-1478.774</td>
<td align="right">15.03336</td>
</tr>
</tbody>
</table>
</div>
<div id="select-significant-snps-from-all-model-type" class="section level2">
<h2>3.2 Select significant SNPs from all model type:</h2>
<p>If the Model type is chosen to be <strong>all</strong> and threshold for <strong>MAF_v</strong> is set to be 0.1, the output for snp_1 with coefficient , P value and log-likelihood function information for <strong>XCI-E</strong>, <strong>XCI</strong> and <strong>XCI-S</strong> respectively,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Covars&lt;-<span class="kw">c</span>(<span class="st">&quot;Age&quot;</span>,<span class="st">&quot;Smoking&quot;</span>,<span class="st">&quot;Treatment&quot;</span>)
SNPs&lt;-<span class="kw">c</span>(<span class="st">&quot;snp_1&quot;</span>,<span class="st">&quot;snp_2&quot;</span>)
output&lt;-<span class="kw">xlink_fit</span>(<span class="dt">os=</span><span class="st">&quot;OS&quot;</span>,<span class="dt">ostime=</span><span class="st">&quot;OS_time&quot;</span>,<span class="dt">snps=</span>SNPs,<span class="dt">gender=</span><span class="st">&quot;gender&quot;</span>,<span class="dt">covars=</span>Covars, <span class="dt">option =</span><span class="kw">list</span>(<span class="dt">type=</span><span class="st">&quot;all&quot;</span>,<span class="dt">MAF_v=</span><span class="fl">0.05</span>),<span class="dt">model=</span><span class="st">&quot;survival&quot;</span>,<span class="dt">data =</span> Rdata)</code></pre></div>
<p>For XCI-E model, snp_1 with coefficient, P value and log-likelihood function information</p>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">Hazard Ratio</th>
<th align="left">Confidence Interval (95%)</th>
<th align="right">P Value</th>
<th align="left">MAF</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>snp_1</td>
<td align="right">1.8293</td>
<td align="left">[1.4411,2.322]</td>
<td align="right">0.0000007</td>
<td align="left">0.20625</td>
</tr>
<tr class="even">
<td>gender</td>
<td align="right">1.0885</td>
<td align="left">[0.8477,1.3978]</td>
<td align="right">0.5061374</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td>Age</td>
<td align="right">1.0232</td>
<td align="left">[1.0157,1.0306]</td>
<td align="right">0.0000000</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td>Smoking</td>
<td align="right">1.3058</td>
<td align="left">[1.0211,1.67]</td>
<td align="right">0.0334504</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td>Treatment</td>
<td align="right">1.2063</td>
<td align="left">[0.9543,1.5249]</td>
<td align="right">0.1167569</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<table>
<thead>
<tr class="header">
<th align="right">Baseline</th>
<th align="right">Full model</th>
<th align="right">Loglik ratio</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">-1493.808</td>
<td align="right">-1482.332</td>
<td align="right">11.47576</td>
</tr>
</tbody>
</table>
<p>For XCI model, snp_1 with coefficient, P value and log-likelihood function information</p>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">Hazard Ratio</th>
<th align="left">Confidence Interval (95%)</th>
<th align="right">P Value</th>
<th align="left">MAF</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>snp_1</td>
<td align="right">1.6392</td>
<td align="left">[1.3895,1.9339]</td>
<td align="right">0.0000000</td>
<td align="left">0.20625</td>
</tr>
<tr class="even">
<td>gender</td>
<td align="right">0.9529</td>
<td align="left">[0.7557,1.2015]</td>
<td align="right">0.6834041</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td>Age</td>
<td align="right">1.0228</td>
<td align="left">[1.0155,1.0301]</td>
<td align="right">0.0000000</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td>Smoking</td>
<td align="right">1.3001</td>
<td align="left">[1.0172,1.6617]</td>
<td align="right">0.0360414</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td>Treatment</td>
<td align="right">1.2356</td>
<td align="left">[0.9781,1.561]</td>
<td align="right">0.0760446</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<table>
<thead>
<tr class="header">
<th align="right">Baseline</th>
<th align="right">Full model</th>
<th align="right">Loglik ratio</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">-1493.808</td>
<td align="right">-1478.774</td>
<td align="right">15.03336</td>
</tr>
</tbody>
</table>
<p>For XCI-S model, snp_1 with coefficient , log-likelihood function information and <strong>gamma</strong> estimation</p>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">Hazard Ratio</th>
<th align="left">Confidence Interval (95%)</th>
<th align="right">P Value</th>
<th align="left">MAF</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>snp_1</td>
<td align="right">1.6596</td>
<td align="left">[1.4031,1.9629]</td>
<td align="right">0.0000000</td>
<td align="left">0.20625</td>
</tr>
<tr class="even">
<td>gender</td>
<td align="right">0.9277</td>
<td align="left">[0.7361,1.1692]</td>
<td align="right">0.5250786</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td>Age</td>
<td align="right">1.0228</td>
<td align="left">[1.0155,1.0301]</td>
<td align="right">0.0000000</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td>Smoking</td>
<td align="right">1.2989</td>
<td align="left">[1.0162,1.6602]</td>
<td align="right">0.0367447</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td>Treatment</td>
<td align="right">1.2374</td>
<td align="left">[0.9795,1.5632]</td>
<td align="right">0.0741144</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<table>
<thead>
<tr class="header">
<th align="right">Baseline</th>
<th align="right">Full model</th>
<th align="right">Loglik ratio</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">-1493.808</td>
<td align="right">-1478.709</td>
<td align="right">15.09833</td>
</tr>
</tbody>
</table>
<table>
<thead>
<tr class="header">
<th align="right">Gamma</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">0.8707407</td>
</tr>
</tbody>
</table>
<p>The best model for snp_1 among model type XCI-E, XCI and XCI-S by using the AIC is</p>
<table>
<thead>
<tr class="header">
<th align="left">Best model by AIC</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">XCI</td>
</tr>
</tbody>
</table>
</div>
<div id="output-for-the-significant-snps-by-p-value" class="section level2">
<h2>3.3 Output for the significant SNPs by P value:</h2>
<p>By setting the threshold for <strong>pv_thold</strong>, the select output become</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Covars&lt;-<span class="kw">c</span>(<span class="st">&quot;Age&quot;</span>,<span class="st">&quot;Smoking&quot;</span>,<span class="st">&quot;Treatment&quot;</span>)
SNPs&lt;-<span class="kw">c</span>(<span class="st">&quot;snp_1&quot;</span>,<span class="st">&quot;snp_2&quot;</span>,<span class="st">&quot;snp_3&quot;</span>)
result&lt;-<span class="kw">xlink_fit</span>(<span class="dt">os=</span><span class="st">&quot;OS&quot;</span>,<span class="dt">ostime =</span><span class="st">&quot;OS_time&quot;</span>,<span class="dt">snps=</span>SNPs,<span class="dt">gender =</span><span class="st">&quot;gender&quot;</span>,<span class="dt">covars=</span>Covars, 
                   <span class="dt">option =</span><span class="kw">list</span>(<span class="dt">type=</span><span class="st">&quot;all&quot;</span>,<span class="dt">MAF_v=</span><span class="fl">0.05</span>), <span class="dt">model=</span><span class="st">&quot;survival&quot;</span>, <span class="dt">data =</span> Rdata)
<span class="kw">select_output</span>(<span class="dt">input=</span>result,<span class="dt">pv_thold=</span><span class="dv">10</span><span class="op">^-</span><span class="dv">5</span>)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">SNP</th>
<th align="right">Hazard Ratio</th>
<th align="left">Confidence Interval (95%)</th>
<th align="right">P Value</th>
<th align="left">MAF</th>
<th align="left">Best model</th>
<th align="right">Gamma</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">snp_1</td>
<td align="right">1.6392</td>
<td align="left">[1.3895,1.9339]</td>
<td align="right">0</td>
<td align="left">0.20625</td>
<td align="left">XCI</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">snp_3</td>
<td align="right">1.5596</td>
<td align="left">[1.3661,1.7805]</td>
<td align="right">0</td>
<td align="left">0.36375</td>
<td align="left">XCI-S</td>
<td align="right">1.538163</td>
</tr>
</tbody>
</table>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
